android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java

```diff
@@ -422,9 +436,29 @@ public boolean handleIterableURL(@NonNull Uri uri, @NonNull IterableActionContex
         return false;
     }
 
+    @Override
+    public String onAuthTokenRequested() {
+        IterableLogger.printInfo();
+
+        try {
+            authHandlerCallbackLatch = new CountDownLatch(1);
+            sendEvent(EventName.handleAuthTokenRequestedCalled.name(), null);
+            authHandlerCallbackLatch.await(30, TimeUnit.SECONDS);
+            authHandlerCallbackLatch = null;
+            return passedAuthToken;
+        } catch (InterruptedException e) {
+            IterableLogger.e(TAG, "auth handler module failed");
+            return null;
+        }
+    }
+
     // ---------------------------------------------------------------------------------------
     // endregion
 
+    public void passAlongAuthToken(String authToken) {
+        passedAuthToken = authToken;
```

Comment:
Need to add something like this here
```
if (authHandlerCallbackLatch != null) {
    authHandlerCallbackLatch.countDown();
}
```

Similar to what we do in `setInAppShowResponse`.